pkg/cmd/api/api_test.go

```diff
@@ -278,6 +278,20 @@ func Test_apiRun(t *testing.T) {
 			stdout: `{"errors": [{"message":"AGAIN"}, {"message":"FINE"}]}`,
 			stderr: "gh: AGAIN\nFINE\n",
 		},
+		{
+			name: "GraphQL string error ",
+			options: ApiOptions{
+				RequestPath: "graphql",
+			},
+			httpResponse: &http.Response{
+				StatusCode: 200,
+				Body:       ioutil.NopCloser(bytes.NewBufferString(`{"errors": ["AGAIN", "FINE"]}`)),
+				Header:     http.Header{"Content-Type": []string{"application/json; charset=utf-8"}},
+			},
+			err:    cmdutil.SilentError,
+			stdout: `{"errors": ["AGAIN", "FINE"]}`,
```

Comment:
GraphQL error message are always guaranteed to be in the `{"message": "..."}` format. If you look closely at the original issue you were aiming to fix, the error response in question is from a REST API, not a GraphQL one.

So I think your direction to fix it is in general good, it's just that this test (and the PR description) needs to be reworded to indicate that it's covering a REST response!

Reply:
Agreed! and also the commit message.